File: 3. Longest Substring Without Repeating Characters/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        head = 0
        maxLength = 0
        seen = {}
        length = len(s)
        for tail in range(length):
            while(s[tail] in seen):
                seen.pop(s[head])
                head += 1
            
            if s[tail] not in seen:
                maxLength = (tail - head + 1) if (tail - head + 1) > maxLength else maxLength
                seen[s[tail]] = tail
                
        return maxLength

File: 3 Longest Substring Without Repeating Characters/3.c
int lengthOfLongestSubstring(char * s){
    int seen[130] = {0};
    if (s[0] == '\0')
	    return 0;
    
    int maxLength = 0;
    char *head = s;
    char *tail = s;
    while (*tail!='\0')
    {
        while(seen[*tail] != 0)
        {
            seen[*head] = 0;
            head ++;
        }
            
        maxLength = (tail - head + 1) > maxLength ? (tail - head + 1) : maxLength;
        seen[*tail] = maxLength;
        tail++;
    }
    return maxLength;
}